core/bindings.rs

```diff
@@ -189,6 +192,29 @@ pub fn initialize_context<'s>(
     get_proxy_details_val.into(),
   );
 
+  // Promise hooks
+  let set_promise_hook_key = v8::String::new(scope, "setPromiseHook").unwrap();
+  let set_promise_hook_tmpl =
+    v8::FunctionTemplate::new(scope, set_promise_hook);
+  let set_promise_hook_val = set_promise_hook_tmpl.get_function(scope).unwrap();
+  core_val.set(
+    scope,
+    set_promise_hook_key.into(),
+    set_promise_hook_val.into(),
+  );
+
+  for (i, hook_name) in
+    ["INIT", "RESOLVE", "BEFORE", "AFTER"].iter().enumerate()
```

Comment:
```suggestion
  use v8::PromiseHookType as Ty;
  for (hook_name, value) in
    [("INIT", Ty::Init), ("RESOLVE", Ty::Resolve), ("BEFORE", Ty::Before), ("AFTER", Ty::After)].iter()
```
(and maybe sort them)

If the original name is good enough you can also do:
```rust
for v in [Ty::After, Ty::Before, Ty::Init, Ty::Resolve].iter() {
  let name = format!("{:?}", v);
  // ...
}
```